core/src/main/java/org/apache/calcite/rex/RexPermuteInputsShuttle.java

```diff
@@ -54,14 +55,41 @@ public class RexPermuteInputsShuttle extends RexShuttle {
   public RexPermuteInputsShuttle(
       Mappings.TargetMapping mapping,
       RelNode... inputs) {
-    this(mapping, fields(inputs));
+    this(mapping, fields(inputs), false);
+  }
+
+  /**
+   * Creates a RexPermuteInputsShuttle with target type matching control.
+   *
+   * <p>The mapping provides at most one target for every source. If a source
+   * has no targets and is referenced in the expression,
+   * {@link org.apache.calcite.util.mapping.Mappings.TargetMapping#getTarget(int)}
+   * will give an error. Otherwise the mapping gives a unique target.
+   *
+   * @param mapping          Mapping that defines source-to-target relation
+   * @param shouldMatchTargetType If true, enforces use target type when remapping inputs
```

Comment:
Based on this description, this should be called 'matchTargetType'. There is no need for 'should'.
I would describe what this does for both 'true' and 'false' cases.
In the 'false' case each input ref is substituted with a reference with the same type as the original input.
In the 'true' case each input ref is substituted with a reference with the same type as the target type.

Reply:
I modified the description of `matchTargetType` according to your suggestion, and kept the other descriptions unchanged. Do you think it's ok?